dockstore-client/src/main/java/io/dockstore/client/cli/nested/WorkflowClient.java

```diff
@@ -377,10 +377,13 @@ public File downloadTargetEntry(String toolpath, ToolDescriptor.TypeEnum type, b
             workflowsApi.getApiClient().setDebugging(Client.DEBUG.get());
             File zipFile = new File(directory, zipFilename(workflow));
             FileUtils.writeByteArrayToFile(zipFile, arbitraryURL, false);
+
+            // If we unzip the file, we can provide a path to the primary descriptor, otherwise just provide a path to the zip file
             if (unzip) {
                 SwaggerUtility.unzipFile(zipFile, directory);
+                return new File(directory, first.get().getWorkflowPath());
             }
-            return new File(directory, first.get().getWorkflowPath());
+            return zipFile;
```

Comment:
This comment for line 354 -- JavaDoc should reflect that it can return a path to the zip file.

Reply:
Added: https://github.com/dockstore/dockstore-cli/pull/108/commits/2ff233d41b72c36e144948f8cbe7271760a5965a